modules/monitoring-applying-a-custom-configuration-to-alertmanager-for-user-defined-alert-routing.adoc

```diff
@@ -0,0 +1,93 @@
+// Module included in the following assemblies:
+//
+// * monitoring/managing-alerts.adoc
+
+:_content-type: PROCEDURE
+[id="applying-a-custom-configuration-to-alertmanager-for-user-defined-alert-routing_{context}"]
+= Applying a custom configuration to Alertmanager for user-defined alert routing 
+
+If you have enabled a separate instance of Alertmanager dedicated to user-defined alert routing, you can overwrite the configuration for this instance of Alertmanager by editing the `alertmanager-user-workload` secret in the `openshift-user-workload-monitoring` namespace.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Print the currently active Alertmanager configuration into the file `alertmanager.yaml`:
++
+[source,terminal]
+----
+$ oc -n openshift-user-workload-monitoring get secret alertmanager-user-workload --template='{{ index .data "alertmanager.yaml" }}' | base64 --decode > alertmanager.yaml
+----
++
+. Edit the configuration in `alertmanager.yaml`:
++
+[source,yaml]
+----
+global:
+  resolve_timeout: 5m
+route:
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 12h
+  receiver: default
+  routes:
+  - match:
+      alertname: Watchdog
+    repeat_interval: 5m
+    receiver: watchdog
+  - match:
+      service: <your_service> <1>
+    routes:
+    - match:
+        <your_matching_rules> <2>
+      receiver: <receiver> <3>
+receivers:
+- name: default
+- name: watchdog
+- name: <receiver>
+#  <receiver_configuration>
+----
+<1> `service` specifies the service that fires the alerts.
+<2> `<your_matching_rules>` specifies the target alerts.
+<3> `receiver` specifies the receiver to use for the alert.
++
+The following Alertmanager configuration example configures PagerDuty as an alert receiver:
++
+[source,yaml,subs=quotes]
+----
+global:
+  resolve_timeout: 5m
+route:
+  group_wait: 30s
+  group_interval: 5m
+  repeat_interval: 12h
+  receiver: default
+  routes:
+  - match:
+      alertname: Watchdog
+    repeat_interval: 5m
+    receiver: watchdog
+  *- match:
+      service: example-app
+    routes:
+    - match:
+        severity: critical
+      receiver: team-frontend-page*
+receivers:
+- name: default
+- name: watchdog
+*- name: team-frontend-page
+  pagerduty_configs:
+  - service_key: "_your-key_"*
+----
++
+With this configuration, alerts of `critical` severity fired by the `example-app` service are sent using the `team-frontend-page` receiver. Typically, these types of alerts are paged to an individual or a critical response team.
```

Comment:
I think that we can keep only one concrete example of modified Alertmanager configuration and remove this part.

Reply:
removed